controllers/secretagentconfiguration_controller.go

```diff
@@ -274,6 +285,14 @@ specSecretsLoop:
 			// if we reach this point, we need to update the k8s secret with new keys
 			k8sApplyRequired = true
 		} // end of keys loop
+		// if redoKeys is true, rerun the keys loop
+		if redoKeys {
+			log.V(1).Info("retrying all keys from secret",
+				"secret_name", secretReq.Name)
+			redoKeys = false
+			// HOTFIX: This is bad practice. We shoudln't use `goto`. The reconcile loop must be refactored asap.
```

Comment:
Added this comment to the code. The reconcile loop is now too long and we should split it into functions (or any other fancier solution). See #150